go/saltpack/encrypt.go

```diff
@@ -30,11 +31,6 @@ type encryptStream struct {
 
 func (es *encryptStream) Write(plaintext []byte) (int, error) {
 
-	if !es.didHeader {
-		es.didHeader = true
-		es.err = es.encoder.Encode(es.header)
-	}
-
```

Comment:
Writing the header here was causing a bug where we never wrote it in the case of an empty message.